### ISSUE TITLE
TypeScript 3.6 Incremental and Build API Support

### ISSUE THREAD

**Issue by kirill-konshin** [zeit/next.js#8702]
Does Next.js support the new APIs of TS 3.6? https://devblogs.microsoft.com/typescript/announcing-typescript-3-6/#apis-to-support-build-and-incremental

It would be extremely useful for large projects, especially ones that use monorepos.

**Comment by zetoke** [comment]
@kirill-konshin The new Nextjs version uses Babel transform plugin to transpile typescript code into JS.
https://babeljs.io/docs/en/babel-plugin-transform-typescript
So it does not really use Typescript's `tsc` compiler.

**Comment by timneutkens** [comment]
Please follow the issue template.

https://github.com/zeit/next.js/issues/new/choose

https://github.com/zeit/next.js/issues/new?template=8.Question_about_next.md

```
# Question about Next.js

GitHub Issues are reserved for Bug reports and Feature requests. The best place to get your question answered is to post it on https://spectrum.chat/next-js.
```

---

Next.js doesn't use `tsc` to compile, it uses the TypeScript Babel preset.

**Comment by kirill-konshin** [comment]
I know that Next.js is based on Babel... this ticket was more of a feature request, sorry I did not follow the template, I thought it's clear enough, I will create a new one.

**Comment by Timer** [comment]
FWIW we already use the incremental API for type checking.

**Comment by kirill-konshin** [comment]
So since #8708 is now closed by @timneutkens let's revive this thread. 

@timneutkens what is the recommendation to build upstream projects? I am well aware that Next.js is using Babel, yet, you have not provided any answer on my comment https://github.com/zeit/next.js/issues/8708#issuecomment-538025408. 

Should I run `tsc --build --noEmit` in parallel with `next start`, is this the only way?

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
